### PR TITLE
[REVIEW] Window Function support for LEAD and LAG statements

### PR DESCRIPTION
**Pull request by Christian8491**
* This PR removes (momentarily) the `ProjectToWindowRule` in order to get constant values correctly from queries which contain CONSTANT values (as LEAD and LAG).
* Updates the parsing at engine level to support successfully Window Functions.
* Add tests for Window Functions which contains LEAD and LAG statement.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

File: tests/BlazingSQLTest/EndToEndTests/windowFunctionTest.py [hunk 1]
```diff
@@ -689,6 +737,245 @@ def executionTest():
...
+                query,
+                queryId,
+                queryType,
+                worder,
+                "",
+                acceptable_difference,
+                use_percentage,
+                fileSchemaType,
+            )
+
```
Comment (williamBlazing) [inline tests/BlazingSQLTest/EndToEndTests/windowFunctionTest.py:978]: Add two more tests:
one where you have some column transformation BEFORE you have window functions and other transformation after the window functions
another one where you have some column transformation in between two window functions that have the same window frame.

**Review by williamBlazing — CHANGES_REQUESTED** [review]
Please create some unit tests for the new complex string parsing we are doing for window functions

File: tests/BlazingSQLTest/EndToEndTests/windowFunctionTest.py [hunk 2]
```diff
@@ -426,6 +424,86 @@ def executionTest():
                 fileSchemaType,
             )
 
+            queryId = "TEST_19"
+            query = """with first_window_order as (
+                            select max(o_totalprice) over
```
Comment (williamBlazing) [inline tests/BlazingSQLTest/EndToEndTests/windowFunctionTest.py:429]: Also add another test that does something like this:

```
                          select  o_orderkey + o_orderpriority,
                           max(o_totalprice) over
                                (
                                    partition by o_orderpriority
                                    order by o_totalprice, o_custkey
                                ) o_max_prices,
                               o_custkey + o_totalprice, 
                                min(o_totalprice) over
                                (
                                    partition by o_orderpriority
                                    order by o_totalprice, o_custkey
                                ) o_min_prices,
                                o_custkey - o_totalprice + 5
                             
                            from orders
```

**Review by williamBlazing — CHANGES_REQUESTED** [review]

**Comment by Christian8491** [comment]
rerun tests

**Comment by Christian8491** [comment]
rerun tests

**Pull request merged**